### DISCUSSION TITLE
iframe from Google Drive vs from Onedrive

### DISCUSSION THREAD

**Discussion by z4ch523** [jgraph/drawio#4309] [Q&A] [answered]
I tried to embed a diagram, that is stored in OneDrive as IFrame and saw, that the complete diagram was inserted into to Code-Snippet. If I do the same with the identical diagram stored in Google Drive, the iframe instead contains a reference to the file stored in the cloud.

A sample would be "#Uhttps%3A%2F%2Fdrive.google.com%2Fuc%3Fid%3D1oD5aDAr-YfTVMwaQGi6aweLsmWqcrcaD4nvRy%26export%3Ddownload".

Why is there a fundamental difference in the way the iFrame works? This might also be related to @datdinhquoc's question here:
https://github.com/jgraph/drawio/discussions/2073

Thanks in advance!

**Comment by alderg** [comment, marked as answer]
OneDrive does not support permanent public URLs, so your best bet is to replace the `src` attribute of the iframe with the URL of the editor, replacing `https://app.diagrams.net` with `https://viewer.diagrams.net`, eg.  `<iframe frameborder="0" style="width:100%;height:345px;" src="https://viewer.diagrams.net/#W..."></iframe>`. This will require the user to login to view the diagram.

**Reply by z4ch523** [reply]
Thanks, this solved it. Just to make sure I understood you right - this is a limitation of OneDrive rather than Drawio?

**Reply by alderg** [reply]
Yes, there is no public link that allows users to access the "raw" file data. If you save the file with a `.drawio.png` extension, in some cases, the _"</> Embed"_ funtion in OneDrive seems to generate a link to the image. See https://github.com/jgraph/drawio/discussions/2073 for details.
